packages/foam-vscode/src/features/link-completion.ts

```diff
@@ -199,6 +199,21 @@ export class CompletionProvider
       item.commitCharacters = linkCommitCharacters;
       return item;
     });
+    resources.push(
+      ...this.ws.list().flatMap(resource =>
+        resource.aliases.map(a => {
+          const item = new ResourceCompletionItem(
```

Comment:
I need to double check later by playing with the feature, but will this insert the identifier of the resource, or the alias itself?
I would be concerned about having the alias there, because it introduces a bunch of implications (e.g. what happens if someone changes the alias list in the note).

So my question is whether the alias is just used to lookup the resource in the completion list, or whether it becomes an identifier for the note?

Reply:
I intended to write the alias as the identifier here, just like suggested in the original issue. The alias identifier will be included in the `listByIdentifier`.

It will be quite complex if we want to support refactoring/renaming aliases. But I think that is not a must-have.

My personal reason for this is that I have some notes written in another software (Logseq) that already uses aliases all over the place 😜

Reply:
> It will be quite complex if we want to support refactoring/renaming aliases. But I think that is not a must-have.

We support refactoring of note names, and I would expect aliases to be supported in that.
I agree it's not trivial, that's why I have been hesitant to add this feature in the past.

Using the alias just to identify a note, but then using the identifier when committing the text, could be an option but I reckon the main reason why people use aliases is not to look up a link, but to make it easier to read the text with the link in it? In that case the feature would not add a lot of value.

I am curious to poll the community on this one, what are your thoughts?

Reply:
> We support refactoring of note names, and I would expect aliases to be supported in that.

IIUC, we currently support updating note filenames and sync to wikilinks. It's slightly tricker for alias because it's in the file content and should use the RenameProvider API. We need to parse to the exact alias as well (supporting comma separated string, trimming whitespaces etc). I don't think it's a must have because the alias is a list, users can easily add additional aliases. Doing a global search and replace is also easy.

For reference, both [Logseq](https://discuss.logseq.com/t/add-ability-to-rename-an-alias-and-all-of-its-instances/3983) and [Obsidian](https://forum.obsidian.md/t/automatically-update-links-when-aliases-are-renamed/9323) doesn't support renaming aliases yet.

> Using the alias just to identify a note, but then using the identifier when committing the text, could be an option but I reckon the main reason why people use aliases is not to look up a link, but to make it easier to read the text with the link in it? In that case the feature would not add a lot of value.

It's an option to generate `[[identifier|alias]]`, but it doesn't look as pretty as we don't have WYSIWYG in vscode.

Reply:
I think the `[[identifier|alias]]` option is actually pretty good, as it addresses most points (except the WYSIWYG) I am concerned about, including one I forgot to mention before: `[[alias]]` is not a valid wikilink (it requires Foam metadata to work, and is very brittle).

So let's do what Obsidian does.
To improve UX we can evaluate having custom decorations for links that have aliases - e.g. this is what I do with my regular links to highlight the label:

<img width="375" alt="image" src="https://user-images.githubusercontent.com/457005/174793232-98fe9bab-386f-4650-9d18-770cce16088c.png">

Reply:
Good point, I'll do `[[identifier|alias]]` instead.